apps/mobile/src/components/TradeChecklistFlow/components/CalculateAmountFlow/components/PremiumOrDiscountScreen/index.tsx

```diff
@@ -0,0 +1,71 @@
+import {
+  FooterButtonProxy,
+  HeaderProxy,
+} from '../../../../../PageWithNavigationHeader'
+import {type NavigationProp, useNavigation} from '@react-navigation/native'
+import {type TradeChecklistStackParamsList} from '../../../../../../navigationTypes'
+import {useTranslation} from '../../../../../../utils/localization/I18nProvider'
+import Content from '../../../Content'
+import useSafeGoBack from '../../../../../../utils/useSafeGoBack'
+import PremiumOrDiscountContent from '../../../../../PremiumOrDiscountContent'
+import {
+  applyFeeOnFeeChangeActionAtom,
+  feeAmountAtom,
+  offerTypeAtom,
+} from '../../atoms'
+import {Text} from 'tamagui'
+import {atom, useAtom, useAtomValue, useSetAtom} from 'jotai'
+import PriceTypeIndicator from '../PriceTypeIndicator'
+import {useEffect} from 'react'
+
+const tempFeeAmountAtom = atom<number>(0)
+
+function PremiumOrDiscountScreen(): JSX.Element {
+  const {t} = useTranslation()
+  const goBack = useSafeGoBack()
+  const navigation: NavigationProp<TradeChecklistStackParamsList> =
+    useNavigation()
+
+  const offerType = useAtomValue(offerTypeAtom)
+  const feeAmount = useAtomValue(feeAmountAtom)
+  const applyFeeOnFeeChange = useSetAtom(applyFeeOnFeeChangeActionAtom)
+  const [tempFeeAmount, setTempFeeAmount] = useAtom(tempFeeAmountAtom)
+
+  useEffect(() => {
+    setTempFeeAmount(feeAmount)
```

Comment:
Is fee amount always derived from fee amount? Can `feeAmountAtom` be derived atom?

Reply:
This tempFeeAtom is just used for PremiumOrDiscountScreen, when changing the value there and on save is stored into feeAtom in state. Then in needs to be synced with state value on render so the user can see the previously set value. Maybe I can change it to useFocusEffect Maybe I don't understand what you mean by deriving so pls correct me if I am wrong.

Reply:
I understand it now. This is used to set initial value on render. 

The reason why `tempFeeAmount` value is not stored in simple useState is, that it needs to be passed to `PremiumOrDiscountContent` component. Makes sense, resolving.